src/experiment.rs

```diff
@@ -13,7 +13,7 @@ pub struct Benchmark<'a> {
 }
 
 impl<'a> Benchmark<'a> {
-    pub fn new(id: &Path) -> Benchmark<'a> {
+    pub fn new(id: &str) -> Benchmark<'a> {
```

Comment:
I was sort-of expecting this to be something like `pub fn new<P: AsRef<Path>>(id: P)` so that users can pass `&str` or `Path` in.

Reply:
Ok, I'll change it.

Reply:
Ah, just seen your comment. I think `AsRef` is preferable because sometimes people will have had to use `Path` manipulation to work out where the benchmark is. `AsRef` makes things a bit easier to use in that case without any downsides that I'm aware of (but I might be wrong there).

Reply:
It looks like this now:
```rust
    pub fn new<P: ?Sized + AsRef<OsStr>>(id: &P) -> Benchmark<'a>
```

I couldn't use `AsRef<Path>`, because `PathBuf` doesn't implement `From<Path>` (only `From<OsStr>`).

Reply:
> Ah, just seen your comment. I think `AsRef` is preferable because sometimes people will have had to use `Path` manipulation to work out where the benchmark is. `AsRef` makes things a bit easier to use in that case without any downsides that I'm aware of (but I might be wrong there).

Makes sense!